mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -0,0 +1,251 @@
+    extract_api_info_for_service,
+    extract_all_api_info_for_service,
+    _REST_API_PATH_PREFIX,
+)
+from mlflow.store.model_registry.rest_store import BaseRestStore
+from mlflow.store._unity_catalog.registry.utils import (
+    registered_model_from_uc_proto,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_ALL_INFO = extract_all_api_info_for_service(
+    UcModelRegistryService, _REST_API_PATH_PREFIX
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _require_arg_unspecified(arg_name, arg_value, default_value=None, message=None):
+    if arg_value != default_value:
+        _raise_unsupported_arg(arg_name, message)
+
+
+def _raise_unsupported_arg(arg_name, message=None):
+    messages = [
+        f"Argument {arg_name} is unsupported for models in the Unity Catalog.",
+    ]
+    if message is not None:
+        messages.append(message)
+    messages.append("See the user guide for more information")
+    raise MlflowException(" ".join(messages))
+
+
+def _raise_unsupported_method(method, message=None):
+    messages = [
+        f"Method {method} is unsupported for models in the Unity Catalog.",
+    ]
+    if message is not None:
+        messages.append(message)
+    messages.append("See the user guide for more information")
+    raise MlflowException(" ".join(messages))
+
+
+def _get_response_from_method(method):
+    method_to_response = {
+        CreateRegisteredModelRequest: CreateRegisteredModelResponse,
+        UpdateRegisteredModelRequest: UpdateRegisteredModelResponse,
+        DeleteRegisteredModelRequest: DeleteRegisteredModelResponse,
+        CreateModelVersionRequest: CreateModelVersionResponse,
+        FinalizeModelVersionRequest: FinalizeModelVersionResponse,
+        UpdateModelVersionRequest: UpdateModelVersionResponse,
+        DeleteModelVersionRequest: DeleteModelVersionResponse,
+        GetModelVersionDownloadUriRequest: GetModelVersionDownloadUriResponse,
+        SearchModelVersionsRequest: SearchModelVersionsResponse,
+        GetRegisteredModelRequest: GetRegisteredModelResponse,
+        GetModelVersionRequest: GetModelVersionResponse,
+        SearchRegisteredModelsRequest: SearchRegisteredModelsResponse,
+        # pylint: disable=line-too-long
+        GenerateTemporaryModelVersionCredentialsRequest: GenerateTemporaryModelVersionCredentialsResponse,
+    }
+    return method_to_response[method]()
+
+
+def _get_all_endpoints_from_method(method):
+    return _METHOD_TO_ALL_INFO[method]
+
+
+def _get_endpoint_from_method(method):
+    return _METHOD_TO_INFO[method]
+
+
+# TODO: re-enable the abstract-method check in a follow-up PR. It's disabled for now because
+# we haven't yet implemented required model version CRUD APIs in this class
+class UcModelRegistryStore(BaseRestStore):  # pylint: disable=abstract-method
+    """
+    Note:: Experimental: This entity may change or be removed in a future release without warning.
```

Comment:
Can we use the `@experimental` decorator here? https://github.com/mlflow/mlflow/blob/88305d95901d1dba89b9e35fe4bab815af4df6c9/mlflow/utils/annotations.py#L8